core/src/test/java/test/failedreporter/issue2517/FailedDataProviderWithFactoryReporterTest.java

```diff
@@ -0,0 +1,59 @@
+package test.failedreporter.issue2517;
+
+import org.testng.Assert;
+import org.testng.TestNG;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+import org.testng.collections.Lists;
+import test.SimpleBaseTest;
+
+import java.io.File;
+import java.util.List;
+
+public class FailedDataProviderWithFactoryReporterTest extends SimpleBaseTest {
+    private File mTempDirectory;
+
+    @BeforeMethod
+    public void setUp() {
+        mTempDirectory = createDirInTempDir("testng-tmp-" + System.currentTimeMillis() % 1000);
+    }
+
+    @AfterMethod
+    public void tearDown() {
+        deleteDir(mTempDirectory);
+    }
+
+    @Test
+    public void failedMethodWithDataProviderAndFactoryShouldHaveInvocationNumbers() {
+        testFailedReporter(
+                new String[]{"f1"},
+                "<include name=\"%s\" invocation-numbers=\"1\"/>",
+                FailedDataProviderWithFactoryReporterSampleTest.class);
+    }
+
+    private void testFailedReporter(String[] expectedMethods, String expectedLine, Class<?>... cls) {
+        triggerTest(cls);
+        runAssertions(mTempDirectory, expectedMethods, expectedLine);
```

Comment:
You can you please remove 

```
static void runAssertions(File outputDir, String[] expectedMethods, String expectedLine)
```

and have `testFailedReporter()` directly invoke

```
runAssertions(outputDir, expectedMethods, expectedLine, 1);
```
That way we dont have additional layers of abstraction.

Reply:
done